storage/src/test/java/tech/pegasys/teku/storage/server/ProtoArrayStorageTest.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.storage.server;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static tech.pegasys.teku.protoarray.ProtoArrayTestUtil.assertThatProtoArrayMatches;
+
+import com.google.common.primitives.UnsignedLong;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.apache.tuweni.bytes.Bytes32;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import tech.pegasys.teku.protoarray.ProtoArray;
+import tech.pegasys.teku.protoarray.ProtoArraySnaphot;
+import tech.pegasys.teku.storage.storageSystem.InMemoryStorageSystem;
+import tech.pegasys.teku.storage.storageSystem.StorageSystem;
+import tech.pegasys.teku.util.async.SafeFuture;
+import tech.pegasys.teku.util.config.StateStorageMode;
+
+public class ProtoArrayStorageTest {
+  private ProtoArrayStorage protoArrayStorage;
+
+  @ParameterizedTest
+  @MethodSource("getStorageSystems")
```

Comment:
(nit) Probably a good idea to use the centralized provider so we have one location where we can update tests that depend on the database:
```suggestion
  @ArgumentsSource(StorageSystemArgumentsProvider.class)
```

Reply:
Done.